internal/execution/handlers.go

```diff
@@ -0,0 +1,158 @@
+package execution
+
+import (
+	"context"
+	"encoding/json"
+	"slices"
+	"strings"
+
+	"github.com/riverqueue/river"
+
+	"github.com/zitadel/zitadel/internal/api/authz"
+	"github.com/zitadel/zitadel/internal/domain"
+	"github.com/zitadel/zitadel/internal/eventstore"
+	"github.com/zitadel/zitadel/internal/eventstore/handler/v2"
+	"github.com/zitadel/zitadel/internal/query"
+	"github.com/zitadel/zitadel/internal/queue"
+	exec_repo "github.com/zitadel/zitadel/internal/repository/execution"
+)
+
+const (
+	HandlerTable = "projections.execution_handler"
+)
+
+type Queue interface {
+	Insert(ctx context.Context, args river.JobArgs, opts ...queue.InsertOpt) error
+}
+
+type Queries interface {
+	TargetsByExecutionID(ctx context.Context, ids []string) (execution []*query.ExecutionTarget, err error)
+	InstanceByID(ctx context.Context, id string) (instance authz.Instance, err error)
+}
+
+type eventHandler struct {
+	eventTypes                 []string
+	aggregateTypeFromEventType func(typ eventstore.EventType) eventstore.AggregateType
+	query                      Queries
+	queue                      Queue
+}
+
+func NewEventHandler(
+	ctx context.Context,
+	config handler.Config,
+	eventTypes []string,
+	aggregateTypeFromEventType func(typ eventstore.EventType) eventstore.AggregateType,
+	query Queries,
+	queue Queue,
+) *handler.Handler {
+	return handler.NewHandler(ctx, &config, &eventHandler{
+		eventTypes:                 eventTypes,
+		aggregateTypeFromEventType: aggregateTypeFromEventType,
+		query:                      query,
+		queue:                      queue,
+	})
+}
+
+func (u *eventHandler) Name() string {
+	return HandlerTable
+}
+
+func (u *eventHandler) Reducers() []handler.AggregateReducer {
+	aggList := make(map[eventstore.AggregateType][]eventstore.EventType)
+	for _, eventType := range u.eventTypes {
+		aggType := u.aggregateTypeFromEventType(eventstore.EventType(eventType))
+		aggEventTypes := aggList[aggType]
+		if !slices.Contains(aggEventTypes, eventstore.EventType(eventType)) {
+			aggList[aggType] = append(aggList[aggType], eventstore.EventType(eventType))
+		}
+	}
+
+	aggReducers := make([]handler.AggregateReducer, len(aggList))
+	i := 0
+	for aggType, aggEventTypes := range aggList {
+		eventReducers := make([]handler.EventReducer, len(aggEventTypes))
+		for j, eventType := range aggEventTypes {
+			eventReducers[j] = handler.EventReducer{
+				Event:  eventType,
+				Reduce: u.reduce,
+			}
+		}
+		aggReducers[i] = handler.AggregateReducer{
+			Aggregate:     aggType,
+			EventReducers: eventReducers,
+		}
+		i++
+	}
+	return aggReducers
+}
+
+func groupsFromEventType(s string) []string {
+	parts := strings.Split(s, ".")
+	partsLen := len(parts)
+	groups := make([]string, partsLen)
+	for i := range parts {
+		groups[(partsLen-1)-i] = strings.Join(parts[:i+1], ".")
+		if i < len(parts)-1 {
+			groups[(partsLen-1)-i] += ".*"
+		}
+	}
+	return groups
```

Comment:
sorry for commenting again I just remembered that there is the reverse function
```suggestion
    parts := strings.Split(s, ".")
	groups := make([]string, len(parts))
	for i := range parts {
		groups[i] = strings.Join(parts[:i+1], ".")
		if i < len(parts)-1 {
			groups[i] += ".*"
		}
	}
    slices.Reverse(groups)
    return groups
```

Reply:
Overlooked, my bad, but good find, didn't now of the reverse function.
Changed.